app/PiCalculator/pi.js

```diff
@@ -0,0 +1,222 @@
+};
+
+function float2int(value) {
+    return value | 0;
+}
+
+var getId = function () {
+    return (((1+Math.random())*0x10000)|0).toString(16).substring(1);
+}
+
+var startTime = getCurrentTime();
+
+if (Meteor.isClient) {
+    var emailValue = getId();
+
+    var startDate = new Date();
+    var prevVal = 0;
+    var counter = 0;
+    var currentProcessingRatePerNSeconds = 0;
+    var startingValue = 0;
+    var perSecondCounter = 0;
+
+    Meteor.subscribe('piData');
+
+    Template.registerHelper('calcPi', function (ptInsideCir, totalNumOfPts) {
+        if (totalNumOfPts === 0) {
+            return 0.0;
+        } else {
+            return (4.0 * ptInsideCir / totalNumOfPts).toFixed(20);
+        }
+    });
+
+    Template.registerHelper('serverPiData', function () {
+            return PIData.findOne({email: 'server', trial: 'server'});
+        }
+    );
+
+    Template.serverPi.helpers({
+        'serverPiData': function () {
+            return PIData.findOne({email: 'server', trial: 'server'});
+        },
+        'calculatePerformanceSpeed': function () {
+            if (perSecondCounter == 0) {
+                startingValue = PIData.findOne({email: "server", trial: "server"}).totalNumOfPts;
+                perSecondCounter++;
+            }
+            var elapsedTime = getCurrentTime() - startTime;
+            return ((PIData.findOne({
+                email: "server",
+                trial: "server"
+            }).totalNumOfPts - startingValue) / elapsedTime).toFixed(2);
+        },
+        'calculatePointsComputedEveryNSeconds': function () {
+            if (counter == 0) {
+                prevVal = PIData.findOne({email: "server", trial: "server"}).totalNumOfPts;
+                counter++;
+            }
+
+            var currentDate = new Date();
+
+            if (currentDate - startDate > n * 1000) {
+                var currentVal = PIData.findOne({email: "server", trial: "server"}).totalNumOfPts;
+                currentProcessingRatePerNSeconds = currentVal - prevVal;
+                startDate = currentDate;
+                prevVal = currentVal
+            }
+
+            console.log("currentProcessingRatePerNSeconds = " + currentProcessingRatePerNSeconds);
+            return currentProcessingRatePerNSeconds;
+        }
+    });
+
+    Template.piTrials.helpers({
+        'piTrialsData': function () {
+            console.log("I'm try trying to retrieve PI Data");
+            pidata = PIData.find({email: {$ne: 'server'}, trial: {$ne: 'server'}});
+            console.log("The pi data has returned");
+            return pidata;
+        },
+        'getProfilePicture': function () {
+            if (signedInWithFacebook) {
+                return "http://graph.facebook.com/" + Meteor.user().services.facebook.id + "/picture/?type=small";
+            }
+            var random = float2int(Math.random() * 4 + 1);
+
+            if (initialized == false) {
+                initialized = true;
```

Comment:
It seems that `initialized` is only checked here.  Is this variable strictly necessary?  If not, remove it, but if so, consider changing `if (initialized == false)` to `if (!initialized)`